.github/PULL_REQUEST_TEMPLATE/release.md

```diff
@@ -0,0 +1,24 @@
+
+### Notes, contributors, tests
+
+- [ ] Review the release notes content and update the text as needed
+- [ ] Write release notes for any essential changes which were missed
+- [ ] Verify the rendering using `make docs` or the ``docs/readthedocs.org:tmt`` check
+- [ ] Use `./scripts/list-new-contributors` to identify new contributors, update `docs/overview.rst` accordingly
+- [ ] Commit and push all release notes and contributor changes
+- [ ] Make sure that the full test coverage has been successfully executed
+
+### Tag, release, downstream
+
+- [ ] Tag the commit with `x.y.z`, push tags using `git push --tags`
```

Comment:
```suggestion
- [ ] After the release PR is merged, checkout `main` branch and `git pull`. *Make sure you have the release commit*
- [ ] Tag the commit with `x.y.z`, push tags using `git push --tags`
```
I guess we were bitten by this a few times, could be worth being excruciatingly verbose 😅